openfisca_core/legislationsxml.py

```diff
@@ -3,926 +3,198 @@
+
+    for key, value in xml_node.attrib.iteritems():
         if key == 'code':
-            pass
-        elif key in ('tail', 'text'):
-            comments.append(value)
-        elif key == 'TRANCHE':
-            scale_json['brackets'] = [
-                transform_bracket_xml_json_to_json(item)
-                for item in value
-                ]
+            name = value
+        elif key == 'format':
+            if value == 'bool':
+                json_node['format'] = 'boolean'
+                value_format = bool
+            elif value == 'percent':
+                json_node['format'] = 'rate'
+            else:
+                json_node['format'] = value
         elif key == 'type':
-            scale_json['unit'] = json_unit_by_xml_json_type.get(value, value)
-        else:
-            scale_json[key] = value
-    if comments:
-        scale_json['comment'] = u'\n\n'.join(comments)
-    return scale_xml_json['code'], scale_json
-
-
-def transform_value_xml_json_to_json(value_xml_json, xml_json_value_to_json_transformer):
-    comments = []
-    value_json = collections.OrderedDict()
-    for key, value in value_xml_json.iteritems():
-        assert key not in ('code', 'format', 'type')
-        if key == 'deb':
-            value_json['start'] = value
-        elif key in ('tail', 'text'):
-            comments.append(value)
-        elif key == 'valeur':
-            try:
-                value_json['value'] = xml_json_value_to_json_transformer(value)
-            except TypeError:
-                log.error(u'Invalid value: {}'.format(value))
-                raise
+            json_node['unit'] = json_unit_by_xml_json_type.get(value)
+        elif key == 'conflicts':
+            json_node['conflicts'] = value.split(',')
+        elif key in {'description', 'origin', 'option'}:
+            json_node[key] = value
         else:
-            value_json[key] = value
-    if comments:
-        value_json['comment'] = u'\n\n'.join(comments)
-    return value_json
-
-
-def transform_end_xml_json_to_json(end_xml_json):
-    comments = []
-    end_json = collections.OrderedDict()
-    for key, value in end_xml_json.iteritems():
-        assert key not in ('code', 'format', 'type')
-        if key == 'deb':
-            end_json['start'] = value
-        elif key in ('tail', 'text'):
-            comments.append(value)
-        else:
-            end_json[key] = value
-    if comments:
-        end_json['comment'] = u'\n\n'.join(comments)
-    return end_json
-
-
-def transform_values_holder_xml_json_to_json(values_holder_xml_json):
-    values = [
-        transform_value_xml_json_to_json(item, float)
-        for item in values_holder_xml_json['VALUE']
-        ]
-
-    if 'END' in values_holder_xml_json:
-        ends = [
-            transform_end_xml_json_to_json(item)
-            for item in values_holder_xml_json['END']
-            ]
+            raise ValueError('Unknown attribute {} : {}'.format(key, value))
```

Comment:
Python 2 warning: string in exception must be 'utf-8' encoded, so:
```py
raise ValueError(u'Unknown attribute {}: {}'.format(key, value).encode('utf-8'))
```

(There is hope: in Python 3 this constraint disappears.)